includes/vendor/MyParcelNL/Sdk/src/Helper/MyParcelCollection.php

```diff
@@ -609,17 +609,34 @@ public function getUserAgent()
     }
 
     /**
-     * @param string $platform
-     * @param string $version
+     * @param string      $platform
+     * @param string|null $version
      *
      * @return self
      * @internal param string $user_agent
+     * @deprecated Use setCustomUserAgent instead
      */
     public function setUserAgent($platform, $version = null)
     {
-       $this::$user_agent = 'MyParcel-' . $platform;
+        $this::$user_agent = 'MyParcel-' . $platform;
         if ($version !== null) {
-           $this::$user_agent .= '/' . str_replace('v', '', $version);
+            $this::$user_agent .= '/' . str_replace('v', '', $version);
+        }
+
+        return $this;
+    }
+
+    /**
+     * @param array $userAgent
+     *
+     * @return self
+     * @internal param string $user_agent
+     */
+    public function setCustomUserAgent(array $userAgent): self
+    {
+        foreach ($userAgent as $key => $value) {
+            $this::$user_agent .= $key;
+            $this::$user_agent .= '/' . str_replace('v', '', $value);
```

Comment:
ben echt geen fan van strings concatten en die replace is misschien gevaarlijk, ik zou het zo doen:

```php
/**
 * @param  array $userAgentMap
 *
 * @return self
 */
public function setUserAgentArray(array $userAgentMap): self
{
    $userAgentMap[] = ['MyParcel-SDK' => self::$version]; // maar dan hoe je die versie echt ophaalt

    $userAgents = [];

    foreach ($userAgentMap as $key => $value) {
        if (Str::startsWith($value, 'v')) {
            $value = str_replace('v', '', $value);
        }

        $userAgents[] = $key . '/' . $value;
    }

    self::$user_agent = implode(' ', $userAgents);

    return $this;
}
```